Include/object.h

```diff
@@ -194,7 +194,9 @@ struct _object {
 
     PyTypeObject *ob_type;
     // VeronaPy: Field used for tracking which region this objects is stored in.
-    // Bottom bits stolen for distinguishing types of region ptr.
+    // Stolen bottom bits:
+    // 1. Indicates the region type. A set flag indicates the immutable region.
+    // 2. This flag is used for object traversal to indicate that it was visited.
```

Comment:
`add_to_region` doesn't need this flag, but the `clean_region` code I'm working on will need it. Tobias and I have discussed some ways to avoid this, but it would make the system more complicated. For now, I was planning to use the flag while traversing the tree for cleaning and document the other options as later improvement ideas.

I already included it in this PR to make sure that bit is already claimed.

Reply:
Perhaps a minor point but I am not happy with the name `clean_region`. We aren't cleaning the _region_ but clearing the dirty flag on the LRC of the region. But perhaps even better would something along the line of "check sendable". Especially if we start with using this as the main driver to ensure region isolation and not the write barrier. Then there might not even be an LRC to clean for a long time.